src/main/java/de/tum/in/www1/artemis/web/rest/ModelingExerciseResource.java

```diff
@@ -116,17 +116,18 @@ public ModelingExerciseResource(ModelingExerciseRepository modelingExerciseRepos
      * PUT /modeling-exercises : Updates an existing modelingExercise.
      *
      * @param modelingExercise the modelingExercise to update
+     * @param exerciseId the id of the exercise
      * @param notificationText the text shown to students
      * @return the ResponseEntity with status 200 (OK) and with body the updated modelingExercise, or with status 400 (Bad Request) if the modelingExercise is not valid, or with
      *         status 500 (Internal Server Error) if the modelingExercise couldn't be updated
      * @throws URISyntaxException if the Location URI syntax is incorrect
      */
-    @PutMapping("/modeling-exercises")
+    @PutMapping("/modeling-exercises/{exerciseId}")
```

Comment:
Why is the entity id a path variable here? As far as I can see it's normally not included explicitly in the path for our PUT endpoints. Is there a specific reason why it's needed here?

Reply:
The main reason is to conform to the standards defined for REST endpoints. `PUT` endpoints should always include the entity id. I did couple of changes to make it clear that we need it in the code. Now we also check if the modeling exercise exists in the database prior to storing update modeling exercise

Reply:
Oh but then the behavior of the endpoint has been changed by the refactoring since it could be used before for both creating and updating whereas now it can only be used for updating. Did you check if all current usages are still valid with the new behavior?

Reply:
Yes, I checked that the creation and updating of the modeling exercise works properly. I also decided to change the update for the modeling submission. This endpoint is a bit more dangerous so I will still have to test it.

Reply:
**UPDATE**
Finished with the testing. Can you check it now?

Reply:
This makes our whole REST API inconsistent. I think we should revert this change!